Branch-SDK/Branch-SDK/BNCLinkCache.m

```diff
@@ -0,0 +1,41 @@
+//
+//  BNCLinkCache.m
+//  Branch-SDK
+//
+//  Created by Qinwei Gong on 1/23/15.
+//  Copyright (c) 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BNCLinkCache.h"
+
+@implementation BNCLinkCache
+
+- (id)init {
+    self = [super init];
+    if (self) {
+        self.cache = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (void)setObject:(NSString *)anObject forKey:(BNCLinkData *)aKey {
+    for (id key in [self.cache allKeys]) {
```

Comment:
True. With that in mind, is there a reason to implement this as a dictionary like structure? An array might be a better structures, since really all you need is an array of the hashes.